codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java

```diff
@@ -238,82 +172,26 @@ public T getAndRemove(T name, T defaultValue) {
     @Override
     public List<T> getAll(T name) {
         checkNotNull(name, "name");
-        List<T> values = new ArrayList<T>(4);
-        int h = hashCodeGenerator.generateHashCode(name);
+
+        LinkedList<T> values = new LinkedList<T>();
```

Comment:
let's keep the `ArrayList`. `LinkedList` just uses more memory. A 10 element ArrayList is the same size, as a two element LinkedList.